src/xref2/env.ml

```diff
@@ -272,35 +272,35 @@ let add_type identifier t env =
   |> add_cdocs identifier t.doc
   |> List.fold_right (add_cdocs identifier) docs
 
-let add_module_type identifier t env =
+let add_module_type identifier (t:Component.ModuleType.t) env =
```

Comment:
Why are these needed ? There's an mli

Reply:
The existence of a mli is irrelevant? The `doc` field is not in scope, the previous code was relying on the specific order of type propagation to resolve the field, this was brittle. Adding a type annotation on `t` fixes the issue.

Reply:
Indeed.